cap_client/cap_api.py

```diff
@@ -412,19 +412,23 @@ def upload_file(self, pid=None, filepath=None, yes=False,
                             '%b-%d-%I%M%p-%G')
                     )
                 make_tarfile(output_filename, filepath)
+                new_tarball = True
                 filepath = output_filename
         else:
             if output_filename is None:
                 output_filename = os.path.basename(filepath)
 
         # data = {'filename': output_filename}
-        return self._make_request(
+        res = self._make_request(
             url="files/{bucket_id}/{filename}".format(
                 bucket_id=bucket_id,
                 filename=output_filename),
             data=open(filepath, 'rb'),
             method='put',
-        )
+            )
+        if new_tarball:
+            os.remove(filepath)
```

Comment:
No error handling here.
If there was any error between creating a tarball and this part, temporary file wont get removed.

Reply:
I think it needs to be a flag (to know that we created a file).

if I put:
```
        if os.path.exists(filepath):
            os.remove(filepath)
```

then it will remove already existing files that I uploaded

Reply:
Use output_filename.

Reply:
it doesn't work

Reply:
it doesn't work